### PR TITLE
Fix multiple rebel squad/vehicle price bugs

### LINKED ISSUE #2150: [Bug]: Rebel AA trucks have incorrect real cost
### Describe the bug

AA trucks are calculated in the tooltip as 2*staticCrew price + vehSDKTruck price * staticAAteamPlayer price, with what I think is a matching funds check in addFIAsquadHC. However, spawnHCGroup calculates the vehicle price component as simply the price of the spawned vehicle type, which is much lower: vehSDKTruck price for vanilla, or 0 for anything else, as vehSDKAA lacks a price definition.

Not sure on the best fix. vehSDKAA could gain a price definition, but unless that's synchronised with truck + static then that may cause trouble in the future. Vanilla would either need a special case or processing of attached objects.

### How to reproduce

1. Recruit a rebel AA truck in vanilla.
2. Recruit a rebel AA truck in RHS. 

### Version

2.5.3

### Have you altered the code?

No

### What i have changed

_No response_

### Map

Doesn't matter

### What server?

LAN Hosted

### Time bug occured (Server time/UTC)

_No response_

### Mods

_No response_

### Additional context

_No response_

### LINKED ISSUE #2152: [Bug]: addFIAsquadHC behaviour is weird and sometimes broken in the low-money case
### Describe the bug

Two further issues with addFIAsquadHC in cases where faction money is low:
1. AT car and AA truck may only spawn the crew even if you have enough money for the vehicle. Missing an isInfantry check.
2. If you have enough money for a non-vehicle squad, but not enough for the accompanying vehicle, it will throw the error "You do not have enough money for this request" before specifying whether to purchase with a vehicle. The message should be changed in this case.

### How to reproduce

Set your money just high enough to purchase a squad type. Recruit that squad. 

### Version

2.5.3 test

### Have you altered the code?

No

### What i have changed

_No response_

### Map

Doesn't matter

### What server?

LAN Hosted, Community Server 2

### Time bug occured (Server time/UTC)

_No response_

### Mods

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jaj22**
### What type of PR is this.
1. [X] Bug
2. [ ] Change
3. [ ] Enhancement

### What have you changed and why?
Both constructed and single-vehicle AA trucks were being underpriced on purchase due to not calculating truck + static AA costs as the other code did. This PR fixes this for both cases with a horrible hack.    

**Edit:** Additional vehicle price bugs fixed:
- Fixed doubled check price for AA and AT vehicles (missing isInfantry check).
- Fixed AT car check price adding SDKTruck price.
- Fixed ancient bug where zu-23 rebel AA trucks were not refunded on dismissal.
- Fixed bug where vanilla AA trucks were double-refunded.
- Fixed misleading error message when you didn't have enough money to purchase a transport vehicle.
- Fixed wrong price shown for transport vehicle.
- Fixed squad mortars and MGs not being charged for.

### Please specify which Issue this PR Resolves.
closes #2150
closes #2152

### Please verify the following and ensure all checks are completed.
1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [X] No
2. [ ] Yes (Please provide further detail below.)

### CODE REVIEW

**Comment by jaj22** [comment]
Found a bunch more pricing bugs, don't merge yet.

File: A3-Antistasi/REINF/dismissSquad.sqf [hunk 1]
```diff
@@ -67,13 +67,11 @@ private _assignedVehicles =	[];
...
 	{
-		_resourcesFIA = _resourcesFIA + ([(typeOf _veh)] call A3A_fnc_vehiclePrice);
-		if (count attachedObjects _veh > 0) then
+		_resourcesFIA = _resourcesFIA + ([typeOf _veh] call A3A_fnc_vehiclePrice);
 		{
-			_subVeh = (attachedObjects _veh) select 0;
-			_resourcesFIA = _resourcesFIA + ([(typeOf _subVeh)] call A3A_fnc_vehiclePrice);
-			deleteVehicle _subVeh;
-		};
+			_resourcesFIA = _resourcesFIA + ([typeOf _x] call A3A_fnc_vehiclePrice);
```
Comment (HakonRydland) [inline A3-Antistasi/REINF/dismissSquad.sqf:72]: would likely tose a lot of errors in the rpt for undefined vehicle prices (should fix vehiclePrice)

File: A3-Antistasi/REINF/dismissSquad.sqf [hunk 2]
```diff
@@ -42,7 +42,7 @@ private _assignedVehicles =	[];
 		{
 			_hr = _hr + 1;
 			_resourcesFIA = _resourcesFIA + (server getVariable [_x getVariable "unitType",0]);
-			if (!isNull (assignedVehicle _x)) then
+			if (!isNull (assignedVehicle _x) and {isNull attachedTo (assignedVehicle _x)}) then
```
Comment (HakonRydland) [inline A3-Antistasi/REINF/dismissSquad.sqf:45]: blocking quads loaded, correct?

**Review by HakonRydland — CHANGES_REQUESTED** [review]
one small change and a clerification

File: A3-Antistasi/REINF/dismissSquad.sqf (hunk 2 above)
Comment (jaj22) [reply A3-Antistasi/REINF/dismissSquad.sqf:45]: Issue was that the AA launchers on the vanilla AA truck were being counted twice: Once from the assigned list and once from the attached objects. Apparently deleteVehicle is not instant.

File: A3-Antistasi/REINF/dismissSquad.sqf (hunk 1 above)
Comment (jaj22) [reply A3-Antistasi/REINF/dismissSquad.sqf:72]: I figured that function just returned zero, but yeah. The whole vehicle value/price system needs a rework but I'll just check vehFIA for the moment.

If people attach random stuff to HC vehicles then that's their problem if it gets deleted without a refund, I guess.

File: A3-Antistasi/REINF/dismissSquad.sqf (hunk 2 above)
Comment (HakonRydland) [reply A3-Antistasi/REINF/dismissSquad.sqf:45]: oh yea (delete vehicle takes a frame i think to clear references, but despawn is instantly i think

File: A3-Antistasi/REINF/dismissSquad.sqf (hunk 1 above)
Comment (HakonRydland) [reply A3-Antistasi/REINF/dismissSquad.sqf:72]: i mean could just get rid of the error, or change it to verbose

Comment (jaj22) [reply A3-Antistasi/REINF/dismissSquad.sqf]: Hmm. I assume the point of an error there is that you're not supposed to call the function when something doesn't have a valid price. Would need to go through all the uses before changing it.

In dismissSquad we only want to refund FIA stuff anyway. Any leftover objects get dropped on the floor, which is fine.

**Comment by jaj22** [comment]
Changed to only attempt attached-vehicle refunds if they're in vehFIA. Anything else is left alone (or dumped on the ground, if the parent object is deleted).

**Review by HakonRydland — APPROVED** [review]

**Pull request merged**